src/language/concurrency.md

```diff
@@ -16,87 +19,108 @@ prevpage:
-the `Future<String>` completes with either a string value or an error.
+This example event loop is synchronous and runs on a single thread.
+However, most Dart applications need to do more than one thing at a time. For example, a
+client application might need to execute an HTTP request, while also listening
+for a user to tap a button. To handle this, Dart offers many async APIs, like
+[Futures, Streams, and async-await](/language/async). 
+These APIs are built around this event loop.
 
-[`readAsStringSync()`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-io/File/readAsStringSync.html
-[`readAsString()`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-io/File/readAsString.html
+For example, consider making a network request:
 
+```dart
+http.get('https://example.com').then((response) {
+  if (response.statusCode == 200) {
+    print('Success!')'
+  }  
+}
+```
 
-#### Why asynchronous code matters
+When this code reaches the event loop, it immediately calls the
+first clause, `http.get`, and returns a `Future`. 
+It also tells the event loop to hold onto the callback in the `then()` clause
+until the HTTP request resolves. When that happens, it should
+execute that callback, passing the result of the request as an argument.
 
-It matters whether a method is synchronous or asynchronous
-because most apps need to do more than one thing at a time.
+![Figure showing async events being added to an event loop and 
+holding onto a callback to execute later
+.](/assets/img/language/concurrency/async-event-loop.png)
 
-Asynchronous computations are often the result of performing computations
-outside of the current Dart code; 
-this includes computations that don't complete immediately, 
-and where you aren't willing to block your Dart code waiting for the result.
-For example, an app might start an HTTP request,
-but need to update its display or respond to user input
-before the HTTP request completes.
-Asynchronous code helps apps stay responsive.
+This same model is generally how the event loop handles all other 
+asynchronous events in Dart, such as [`Streams`][].
 
-These scenarios include operating system calls like
-non-blocking I/O, performing an HTTP request, or communicating with a browser. 
-Other scenarios include waiting for computations
-performed in another Dart isolate as described below, 
-or maybe just waiting for a timer to trigger. 
-All of these processes either run in a different thread, 
-or are handled by the operating system or the Dart runtime, 
-which allows Dart code to run concurrently with the computation.
+[`Streams`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-async/Stream-class.html
 
+## Asynchrony types and syntax
+
+If you're already familiar with `Future`, `Stream`, and async-await,
+then you can skip ahead to the [isolates section][].
+
+[isolates section]: #isolates
+
+### Future type syntax
+ 
+A promise to eventually provide a string value is typed as `Future<String>`.
```

Comment:
As a reader, I was a bit surprised this jumped straight to an example. I think it would be helpful to offer a more generic introduction before jumping to a specific instance of it.